src/pages/home/report/ReportActionItemParentAction.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import type {WithLocalizeProps} from '@components/withLocalize';
+import type {WindowDimensionsProps} from '@components/withWindowDimensions/types';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as ReportActionsUtils from '@libs/ReportActionsUtils';
+import * as Report from '@userActions/Report';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type * as OnyxTypes from '@src/types/onyx';
+import AnimatedEmptyStateBackground from './AnimatedEmptyStateBackground';
+import ReportActionItem from './ReportActionItem';
+
+type ReportActionItemParentActionOnyxProps = {
+    /** ONYX PROPS */
+
+    /** The report currently being looked at */
+    report: OnyxEntry<OnyxTypes.Report>;
+
+    /** The actions from the parent report */
+    // TO DO: Replace with HOC https://github.com/Expensify/App/issues/18769.
+    parentReportActions: OnyxEntry<OnyxTypes.ReportActions>;
+};
+type ReportActionItemParentActionProps = WithLocalizeProps &
+    WindowDimensionsProps &
+    ReportActionItemParentActionOnyxProps & {
+        /** Flag to show, hide the thread divider line */
+        shouldHideThreadDividerLine: boolean;
+
+        /** Flag to display the new marker on top of the comment */
+        shouldDisplayNewMarker: boolean;
+
+        /** The id of the report */
+        reportID: string;
+
+        /** The id of the parent report */
+        // eslint-disable-next-line react/no-unused-prop-types
+        parentReportID: string;
+    };
+
+function ReportActionItemParentAction({report, parentReportActions, isSmallScreenWidth, shouldHideThreadDividerLine, shouldDisplayNewMarker}: ReportActionItemParentActionProps) {
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    if (!parentReportActions) {
+        return;
+    }
+    if (!report) {
+        return;
+    }
+    const parentReportAction = parentReportActions[`${report?.parentReportActionID}`];
+    const shouldDisableOpacity = parentReportAction?.pendingAction ?? false;
+    const pendingAction = report?.pendingFields?.addWorkspaceRoom ?? report?.pendingFields?.createChat;
+    const errors = report?.errorFields?.addWorkspaceRoom ?? report?.errorFields?.createChat;
```

Comment:
I don't see any reason for saving these into constants. I think we can use them directly as they were previously.

Reply:
I removed the constants.